tree/dataframe/src/RNTupleDS.cxx

```diff
@@ -337,8 +340,14 @@ void RNTupleDS::AddField(const RNTupleDescriptor &desc, std::string_view colName
          valueField =
             std::make_unique<ROOT::Experimental::RArrayAsRVecField>("", std::move(valueField), fieldInfo.fNRepetitions);
       } else {
-         // Actual ROOT::RVec
-         valueField = std::make_unique<ROOT::Experimental::RRVecField>("", std::move(valueField));
+         // Actual collection. std::vector, std::array and RVecs get turned into RVecs. All other collection types keep
```

Comment:
IIUC the `std::array` conversion part is still handled by the previous `if` branch through the `RArrayAsRVecField` right?